### PR TITLE
Navigation js to ts and fix nav issues

### PR DESCRIPTION
**Pull request by Citrullin**
I fixed some of the navigation issues. Still not ideal though. The first element only gets selected when you open the app. Going to another view and the first element isn't selected. But infinite scrolling works fine.

### CODE REVIEW

File: src/machinery/eventListener.ts [hunk 1]
```diff
@@ -14,14 +14,16 @@ let selection = undefined;
 
 const elementSelector = (selectedElement) => {
   selection = selectedElement;
-  console.log('Selected: ' + selectedElement);
 };
-let navigation = new Navigation([], elementSelector);
+let navigation = null;
```
Comment (mehlium) [inline src/machinery/eventListener.ts]: It's good practice to have explicit types for this: `let navigation: Navigation | null = null`

**Review by mehlium — APPROVED** [review]
LGTM! :-) Had one comment

File: src/machinery/eventListener.ts (hunk 1 above)
Comment (Citrullin) [reply src/machinery/eventListener.ts]: Yeahh, true that. But if it is empty there are a couple of problems. I mean I can check it. Okay, I check inside the class, if it is empty.

Comment (Citrullin) [reply src/machinery/eventListener.ts]: @aspic Fixed it.

**Pull request merged**